src/eigen/preconditioners.jl

```diff
@@ -35,8 +34,11 @@ end
 function PreconditionerTPA(basis::PlaneWaveBasis{T}, kpt::Kpoint; default_shift=1) where T
     kinetic_term = [t for t in basis.model.term_types if t isa Kinetic]
     isempty(kinetic_term) && error("Preconditioner should be disabled when no Kinetic term is used.")
-    scaling = only(kinetic_term).scaling_factor
-    kin = Vector{T}([scaling * sum(abs2, q) for q in Gplusk_vectors_cart(basis, kpt)] ./ 2)
+    kinetic_term = only(kinetic_term)
+    scaling = kinetic_term.scaling_factor
+    blowup = kinetic_term.blowup     # blowup for energy cut-off smearing
+    kin = Vector{T}([scaling * basis.Ecut * blowup(norm(q)/√(2*Ecut))
+                     for q in Gplusk_vectors_cart(basis, kpt)])
```

Comment:
sucks that you have to recompute it. Can the Preconditioner maybe get the `HamiltonianBlock` object and use it to get the kinetic energies?

Reply:
Yes I'd argue that makes sense as the preconditioner can be thought of as a black box trying to construct an approximation of the discretised operator.